### PR TITLE
Add a new Shade resource transformer to merge file content (license files)

### PR DESCRIPTION
**Pull request by olamy**

### CODE REVIEW

File: jetty-build-support/src/main/java/org/eclipse/jetty/toolchain/shade/MergeFilesTransformer.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.jar.JarEntry;
+import java.util.jar.JarOutputStream;
+
```
Comment (janbartel) [inline jetty-build-support/src/main/java/org/eclipse/jetty/toolchain/shade/MergeFilesTransformer.java]: A comment here about what this transformer does would be useful. I had to go find the source for ResourceTransformer to work out what it was doing (and ResourceTransformer doesn't have any useful javadoc either :( ).

File: jetty-build-support/src/main/java/org/eclipse/jetty/toolchain/shade/MergeFilesTransformer.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    private boolean preserveResourceContent = true;
+
+    private ByteArrayOutputStream data = new ByteArrayOutputStream();
+
+    public MergeFilesTransformer()
+    {
+        //no op
+    }
+
+    public MergeFilesTransformer( List<String> files, String resource, boolean preserveResourceContent )
```
Comment (janbartel) [inline jetty-build-support/src/main/java/org/eclipse/jetty/toolchain/shade/MergeFilesTransformer.java]: A javadoc comment on the purpose of the args would be very useful in working out how to use this transformer. For example, what are the files argument - are they absolute locations on disk somewhere or are they relative to inside the artifacts that the shade plugin has been configured with?

File: Jenkinsfile [hunk 3]
```diff
@@ -1,7 +1,7 @@
 #!groovy
 
 // if removing jdk look at the test made for deploying
-def jdks = ["jdk8", "jdk9", "jdk10", "jdk11"]
```
Comment (janbartel) [inline Jenkinsfile]: Seems to be a heap of changes in this pr, more than I would have expected to see just for the addition of the MergeFilesTransformer? Should these other changes be in a separate pr?

Comment (joakime) [reply Jenkinsfile]: Agreed, this `Jenkinsfile` change should be a separate commit / PR

File: jetty-osgi-servlet-api/pom.xml [hunk 4]
```diff
@@ -135,12 +135,26 @@
...
                 <transformer implementation="org.apache.maven.plugins.shade.resource.IncludeResourceTransformer">
                   <resource>META-INF/MANIFEST.MF</resource>
-                  <file>target/classes/META-INF/MANIFEST.MF</file>
+                  <file>${project.build.outputDirectory}/META-INF/MANIFEST.MF</file>
                 </transformer>
+                 <transformer implementation="org.eclipse.jetty.toolchain.shade.MergeFilesTransformer">
+                   <resource>META-INF/LICENSE</resource>
+                   <files>
+                     <file>${project.build.directory}/sources/META-INF/LICENSE</file>
+                     <file>${project.build.directory}/sources/META-INF/LICENSE.txt</file>
```
Comment (joakime) [inline jetty-osgi-servlet-api/pom.xml]: Oddly named files.
Is one Eclipse License? the other CDDL? (or something)

Comment (olamy) [reply jetty-osgi-servlet-api/pom.xml]: @joakime yes...
